ProcessLib/PhaseField/PhaseFieldFEM.h

```diff
@@ -0,0 +1,609 @@
+
+            auto const& B = _ip_data[ip].b_matrices;
+
+            auto const& C_tensile = _ip_data[ip].C_tensile;
+            auto const& C_compressive = _ip_data[ip].C_compressive;
+
+            auto const& strain_energy_tensile =
+                _ip_data[ip].strain_energy_tensile;
+            auto const& sigma_tensile = _ip_data[ip].sigma_tensile;
+
+            auto& history_variable = _ip_data[ip].history_variable;
+            auto& history_variable_prev = _ip_data[ip].history_variable_prev;
+
+            auto const& sigma_real = _ip_data[ip].sigma_real;
+
+            // Kdd_1 defines one term which both used in Kdd and local_rhs for
+            // phase field
+            double const gc = _process_data.crack_resistance(t, x_position)[0];
+            double const ls =
+                _process_data.crack_length_scale(t, x_position)[0];
+            typename ShapeMatricesType::NodalMatrixType const Kdd_1 =
+                dNdx.transpose() * 2 * gc * ls * dNdx;
+
+            //
+            // displacement equation, displacement part
+            //
+            double const k = _process_data.residual_stiffness(t, x_position)[0];
+            double const d_ip = N.dot(d);
+            double const degradation = d_ip * d_ip * (1 - k) + k;
+            _ip_data[ip].updateConstitutiveRelation(t, x_position, dt, u,
+                                                    degradation);
+
+            local_Jac.template block<displacement_size, displacement_size>(
+                         displacement_index, displacement_index)
+                .noalias() += B.transpose() *
+                              (degradation * C_tensile + C_compressive) * B * w;
+
+            typename ShapeMatricesType::template MatrixType<
+                DisplacementDim, displacement_size> N_u = ShapeMatricesType::
+                template MatrixType<DisplacementDim, displacement_size>::Zero(
+                    DisplacementDim, displacement_size);
+
+            for (int i = 0; i < DisplacementDim; ++i)
+                N_u.template block<1, displacement_size / DisplacementDim>(
+                       i, i * displacement_size / DisplacementDim)
+                    .noalias() = N;
+
+            auto const rho_sr = _process_data.solid_density(t, x_position)[0];
+            auto const& b = _process_data.specific_body_force;
+            local_rhs.template segment<displacement_size>(displacement_index)
+                .noalias() -=
+                (B.transpose() * sigma_real - N_u.transpose() * rho_sr * b) * w;
+
+            //
+            // displacement equation, phasefield part
+            //
+            Kud.noalias() += B.transpose() * 2 * d_ip * sigma_tensile * N * w;
+
+            double const d_dot_ip = N.dot(d_dot);
+
+            if (history_variable_prev < strain_energy_tensile)
+            {
+                history_variable = strain_energy_tensile;
+                Kdu.noalias() = Kud.transpose();
+            }
+            else
```

Comment:
If you have drastic convergence issues you can decouple the system (and restore convexity) by updating H only once a global increment has converged. Then K_du is always zero. But I guess this would be better done via staggered coupling and controlling the coupling iterations, as in that case monolithic coupling only serves to increase the equation system.
So: Nothing to do here.
@endJunction FYI as an update on today's discussion. Btw, how far is staggered scheme/coupling manager? Processes of immediate interest for sequential coupling are T-(H)M, Phase field.